sphinx/texinputs/sphinx.sty

```diff
@@ -1049,17 +1072,28 @@
        \vskip\spx@abovecaptionskip
        \def\sphinxVerbatim@Before
            {\sphinxVerbatim@Title\nointerlineskip
-            \kern\dimexpr-\dp\strutbox+\sphinxbelowcaptionspace\relax}%
+            \kern\dimexpr-\dp\strutbox+\sphinxbelowcaptionspace
+                 % if no frame (code-blocks inside table cells), remove
+                 % the "verbatimsep" whitespace from the top (better visually)
+                 \ifspx@opt@verbatimwithframe\else-\sphinxverbatimsep\fi
+                 % caption package adds \abovecaptionskip vspace, remove it
+                 \spx@ifcaptionpackage{-\abovecaptionskip}{}\relax}%
```

Comment:
If caption package is active and caption is on top, it adds `\abovecaptionskip` *below* the caption, contrarily to standard LaTeX. We remove it for consistency and then only `\sphinxbelowcaptionspace` governs for all four: tabular, tabulary, longtable, and literalblock the spacing to frame of object.

Reply:
When one uses `caption` package it succeeds in achieving same spacing for positioning of a top caption for both `table` environment (with a tabular) or `longtable`. But Sphinx does not use `table` environment for a `tabular`, I have observed that in such non-table environment, the `caption` package does not succeed in positioning vertically the caption exactly like it does for longtable. Thus, the necessity remains to keep the `\sphinxaftertopcaption` macro (especially because `parskip` package is loaded) and its way of achieving `\sphinxbelowcaptionspace` vertical spacing. So we also need to achieve this for literal blocks, which is done here.

The situation is thus completely different between the status of `\abovecaptionskip` (used with `caption` package between caption and object) and the one of `\belowcaptionskip` (used on external side if `caption` package used). In the former case Sphinx needs special corrective measures for achieving identical result with tabular(y), longtable and literal blocks, in the latter case, it does not have to worry.